core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -432,95 +470,124 @@ angular.module('oppia').directive('questionsList', [
+                  AlertsService.addInfoMessage(
+                    'Skill already linked to question');
+                  return;
+                }
               }
+
+              ctrl.associatedSkillSummaries.push(
+                ShortSkillSummaryObjectFactory.create(
+                  summary.id, summary.description));
+              ctrl.skillLinkageModificationsArray = [];
+              ctrl.skillLinkageModificationsArray.push({
+                id: summary.id,
+                task: 'add'
+              });
             }, function() {
-              ContextService.resetImageSaveDestination();
-              ctrl.editorIsOpen = false;
-              $location.hash(null);
+              // Note to developers:
+              // This callback is triggered when the Cancel button is
+              // clicked. No further action is needed.
             });
           };
 
+          ctrl.updateSkillLinkage = function(commitMsg) {
+            EditableQuestionBackendApiService.editQuestionSkillLinks(
+              ctrl.questionId, ctrl.skillLinkageModificationsArray,
+              ctrl.difficulty).then(
+              data => {
+                $timeout(function() {
+                  QuestionsListService.resetPageNumber();
+                  _reInitializeSelectedSkillIds();
+                  ctrl.getQuestionSummariesAsync(
+                    ctrl.selectedSkillIds, true, true
+                  );
+                  ctrl.editorIsOpen = false;
+                  ctrl.saveAndPublishQuestion(commitMsg);
+                }, 500);
+              });
+          };
+
+          ctrl.saveQuestion = function() {
+            ContextService.resetImageSaveDestination();
+            $location.hash(null);
+            var commitMessage = null;
+            if (ctrl.questionIsBeingUpdated) {
+              $uibModal.open({
+                templateUrl:
+                      UrlInterpolationService.getDirectiveTemplateUrl(
+                        '/components/question-directives' +
+                          '/modal-templates/' +
+                          'question-editor-save-modal.template.html'),
+                backdrop: true,
+                controller: 'ConfirmOrCancelModalController'
+              }).result.then(function(commitMsg) {
+                commitMessage = commitMsg;
+                if (ctrl.skillLinkageModificationsArray &&
+                    ctrl.skillLinkageModificationsArray.length > 0) {
+                  ctrl.updateSkillLinkage(commitMessage);
+                } else {
+                  ContextService.resetImageSaveDestination();
+                  ctrl.saveAndPublishQuestion(commitMessage);
+                }
+              });
+            } else {
+              if (ctrl.skillLinkageModificationsArray.length > 0) {
+                ctrl.updateSkillLinkage(null);
+              } else {
+                ContextService.resetImageSaveDestination();
+                ctrl.saveAndPublishQuestion(null);
```

Comment:
Why are we calling updateSkillLinkage and saveAndPublishQuestion with 'null' commitMessage? Looks weird..

Reply:
Earlier we were sending undefined, null is better for types. Also for the new question, no new commit message is required. Hence null.

Reply:
I see. Thanks for explaining!